src/n_body.rs

```diff
@@ -1,144 +1,224 @@
 // The Computer Language Benchmarks Game
 // http://benchmarksgame.alioth.debian.org/
 //
-// contributed by the Rust Project Developers
-// contributed by Matt Brubeck
+// contributed by the packed_simd developers
+// contributed by Andre Bogus
 // contributed by TeXitoi
 
-const PI: f64 = 3.141592653589793;
+use std::f64::consts::PI;
+use std::ops::*;
+
 const SOLAR_MASS: f64 = 4.0 * PI * PI;
-const YEAR: f64 = 365.24;
-const N_BODIES: usize = 5;
+const DAYS_PER_YEAR: f64 = 365.24;
+
+#[derive(Copy, Clone)]
+pub struct F64x2(f64, f64);
+
+#[derive(Copy, Clone)]
+pub struct F64x4(f64, f64, f64, f64);
+
+impl F64x2 {
+    pub fn splat(x: f64) -> F64x2 { F64x2(x, x) }
+    pub fn new(a: f64, b: f64) -> F64x2 { F64x2(a, b) }
+    pub fn sqrte(self) -> F64x2 { F64x2(self.0.sqrt(), self.1.sqrt()) }
+    pub fn write_to_slice_unaligned(self, slice: &mut [f64]) {
+        slice[0] = self.0;
+        slice[1] = self.1;
+    }
+}
 
-static BODIES: [Planet;N_BODIES] = [
-    // Sun
-    Planet {
-        x: 0.0, y: 0.0, z: 0.0,
-        vx: 0.0, vy: 0.0, vz: 0.0,
-        mass: SOLAR_MASS,
-    },
-    // Jupiter
-    Planet {
-        x: 4.84143144246472090e+00,
-        y: -1.16032004402742839e+00,
-        z: -1.03622044471123109e-01,
-        vx: 1.66007664274403694e-03 * YEAR,
-        vy: 7.69901118419740425e-03 * YEAR,
-        vz: -6.90460016972063023e-05 * YEAR,
-        mass: 9.54791938424326609e-04 * SOLAR_MASS,
-    },
-    // Saturn
-    Planet {
-        x: 8.34336671824457987e+00,
-        y: 4.12479856412430479e+00,
-        z: -4.03523417114321381e-01,
-        vx: -2.76742510726862411e-03 * YEAR,
-        vy: 4.99852801234917238e-03 * YEAR,
-        vz: 2.30417297573763929e-05 * YEAR,
-        mass: 2.85885980666130812e-04 * SOLAR_MASS,
-    },
-    // Uranus
-    Planet {
-        x: 1.28943695621391310e+01,
-        y: -1.51111514016986312e+01,
-        z: -2.23307578892655734e-01,
-        vx: 2.96460137564761618e-03 * YEAR,
-        vy: 2.37847173959480950e-03 * YEAR,
-        vz: -2.96589568540237556e-05 * YEAR,
-        mass: 4.36624404335156298e-05 * SOLAR_MASS,
-    },
-    // Neptune
-    Planet {
-        x: 1.53796971148509165e+01,
-        y: -2.59193146099879641e+01,
-        z: 1.79258772950371181e-01,
-        vx: 2.68067772490389322e-03 * YEAR,
-        vy: 1.62824170038242295e-03 * YEAR,
-        vz: -9.51592254519715870e-05 * YEAR,
-        mass: 5.15138902046611451e-05 * SOLAR_MASS,
-    },
-];
-
-#[derive(Clone, Copy)]
-struct Planet {
-    x: f64, y: f64, z: f64,
-    vx: f64, vy: f64, vz: f64,
-    mass: f64,
+impl F64x4 {
+    pub fn splat(x: f64) -> F64x4 { F64x4(x, x, x, x) }
+    pub fn new(a: f64, b: f64, c: f64, d: f64) -> F64x4 { F64x4(a, b, c, d) }
+    pub fn sum(self) -> f64 {
+        (self.0 + self.1) + (self.2 + self.3)
```

Comment:
Why these parenthesis? to minimize instructions?

Reply:
I found it beneficial to reduce data dependencies. Though the perf increase is admittedly small.